services/motion/builtin/builtin_test.go

```diff
@@ -639,6 +636,63 @@ func TestMoveOnGlobe(t *testing.T) {
 	})
 }
 
+func TestReplanning(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	gpsPoint := geo.NewPoint(0, 0)
+	dst := geo.NewPoint(gpsPoint.Lat(), gpsPoint.Lng()+1e-5)
+	epsilonMM := 15.
+	motionCfg := &motion.MotionConfiguration{PositionPollingFreqHz: 100, ObstaclePollingFreqHz: 1, PlanDeviationMM: epsilonMM}
+
+	t.Run("check we dont replan with a good sensor", func(t *testing.T) {
+		t.Parallel()
+		noise := spatialmath.NewPoseFromPoint(r3.Vector{Y: epsilonMM - 0.1})
+		injectedMovementSensor, _, kb, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, dst, noise)
+
+		moveRequest, err := ms.(*builtIn).newMoveOnGlobeRequest(ctx, kb.Name(), dst, injectedMovementSensor.Name(), nil, motionCfg, nil)
+		test.That(t, err, test.ShouldBeNil)
+
+		ctx, cancel := context.WithTimeout(ctx, time.Second)
+		defer cancel()
+		ma := newMoveAttempt(ctx, moveRequest)
+		ma.start()
+		select {
+		case <-ma.ctx.Done():
+			t.FailNow()
```

Comment:
I recommend logging a `t.Error` before all `t.FailNow` calls b/c b/c I don't know if FailNow gives you a line number of where it failed.